src/js/letters/components/Address.jsx

```diff
@@ -60,6 +60,37 @@ class Address extends React.Component {
     return lowerCity === 'apo' || lowerCity === 'fpo' || lowerCity === 'dpo';
   }
 
+  adjustStateNames(stateList, militaryStates, city) {
```

Comment:
There are 3 different formats the list of states could be in:
1. The list fetched from EVSS is just an array of abbreviations
e.g., `['AL', 'AK', 'AZ'...]`

2. The list of constants we have hard-coded on our side, which is what we decided to actually use, is an object of key-value pairs
e.g., `{AL: 'Alabama', AK: 'Alaska', AZ: 'Arizona'....}`

3. There's the list that `ErrorableSelect` expects, which is an array of objects with 2 keys: `label` and `value`
e.g., `[{label: 'Alabama', value: 'AL'}, {label: 'Alaska', value: 'AK'}....]`

This code transforms list 2 (hard-coded object list) into list 3 (array of objects) so that it can be processed by the `ErrorableSelect` component.

This logic is a little weird. There are basically a few steps I have to take to reformat the state list.
1. Create an empty array
2. Map through the hard-coded state list (list 2) and create a new object for each state, and then push that object into the newly created array
3. Add the military states into that array list
4. Alphabetize the list
5. Filter the list down to the military states if the user has entered a military address city (this is a feature from our common `Address` component across Vets.gov, figured we wanted to maintain that for this app)

I'm having a hard time figuring out how to test this functionality. I figured the important things to test here are to make sure the list is populated and that the military states have been added to the list (could also test the alphabetization and the filtering). So far I tried rendering this component, getting the input for the state dropdown, and then trying to test the values of the options inside that select, but apparently that doesn't work. Any other thoughts?

Reply:
I'm curious how you tried doing this. Did you use `findDOMNode` (or `getFormDOM` from `test/util/schemaform-utils.jsx`)? I think you could do something like

```js
const component = ReactTestUtils.renderIntoDocument( ... );
const tree = findDOMNode(component);

const options = Array.from(tree.querySelectorAll('option'));
expect(options).to.include('APO'); // etc.
// TODO: Test that all the options are in order, etc
```

Reply:
I tried this and unfortunately it's not working for me. The `options` that get rendered are React elements, and I'm not sure how to access their value. Ordinarily doing `.value` on a React element in a test works, but for some reason it doesn't work for the options, only the select itself.